controllers/check_client_compatibility.go

```diff
@@ -42,12 +42,6 @@ func (c checkClientCompatibility) reconcile(_ context.Context, r *FoundationDBCl
 		return nil
 	}
 
-	adminClient, err := r.getDatabaseClientProvider().GetAdminClient(cluster, r)
-	if err != nil {
-		return &requeue{curError: err}
-	}
-	defer adminClient.Close()
-
```

Comment:
Moving it at a later stage where it actually needed. This can help in unnecessary fetching of the admin client if there are error before its actual use,.